stem_cell/differentiator.py

```diff
@@ -1,18 +1,30 @@
 class Differentiator:
     PAYLOAD_ENTRY_ANSIBLE_PLAYBOOK = '/main.yml'
 
-    def __init__(self, user_data_gateway, version_control_system, ansible):
+    def __init__(self, user_data_gateway, version_control_system, ansible, file_gateway):
         self.ansible = ansible
+        self.file_gateway = file_gateway
         self.user_data_gateway = user_data_gateway
         self.version_control_system = version_control_system
 
     def differentiate(self):
-        payload_location = self.clone_differentiation_payload()
+        payload_location = self.clone_payload()
         self.ansible.run_playbook(payload_location + self.PAYLOAD_ENTRY_ANSIBLE_PLAYBOOK)
 
-    def clone_differentiation_payload(self):
+    def clone_payload(self):
+        private_key = self.user_data_gateway.get_version_control_private_key()
+        if private_key:
```

Comment:
My preference is using indentation to show primary and secondary flows within methods.

1 level of indentation generally means "this doesn't happen often"

So there is something wrong with this line, private_key is expected to be the preferred option, but it's dealt with on an indentation level of 1.